napari/plugins/io.py

```diff
@@ -74,3 +68,194 @@ def read_data_with_plugins(
                 # error anyway, so it looks a bit weird to show them that the
                 # "builtin plugin" didn't work.
                 logger.error(err.format_with_contact_info())
+
+
+def write_multiple_layers_with_plugin(
+    path: str, layers: List[Layer], plugin_name: Optional[str] = None
+):
+    """Write data from multiple layers data with a plugin.
+
+    If no `plugin_name` is provided we loops through plugins to find the first
+    one that knows how to handle the combination of layers and is able to write
+    the file. If no plugins offer `napari_get_writer` for that combination of
+    layers then the default `napari_get_writer` will create a folder and call
+    `napari_write_<layer>` for each layer using the `layer.name` variable
+    to modify the path such that the layers are written to unique files in
+    the folder.
+
+    If a `plugin_name` is provided, then we call we call `napari_get_writer`
+    for that plugin, and if it doesn’t return a WriterFunction we error,
+    otherwise we call it and if that fails if it we error.
+
+    Exceptions will be caught and stored as PluginErrors
+    (in plugin_manager._exceptions)
+
+    Parameters
+    ----------
+    path : str
+        The path (file, directory, url) to write.
+    layers : List of napari.layers.Layer
+        List of napari layers to write.
+    """
+    layer_data = [
+        (layer.data, layer._get_state(), layer.__class__.__name__.lower())
+        for layer in layers
+    ]
+    layer_types = [ld[2] for ld in layer_data]
+
+    plugin_manager = get_plugin_manager()
+
+    if plugin_name is None:
+        # Loop through all plugins using first successful one
+        skip_impls = []
+        while True:
+            (writer, implementation) = plugin_manager.hook.napari_get_writer(
+                path=path,
+                layer_types=layer_types,
+                _return_impl=True,
+                _skip_impls=skip_impls,
```

Comment:
@tlambert03  I now get the following warning when using this hookspec (or a similar one for the readers). Not sure how serious it is, but would be good to either fix or suppress
```
/Users/nsofroniew/GitHub/napari/napari/plugins/io.py:116: UserWarning: Argument(s) ('__multicall__',) which are declared in the hookspec can not be found in this hook call
  _skip_impls=skip_impls,
```